include/crypto/cipher_structs.h

```diff
@@ -175,6 +175,8 @@ struct cipher_ctx {
 /* Whether the hardware/driver supports autononce feature */
 #define CAP_AUTONONCE			BIT(7)
 
+/* Don't prefix IV to cipher blocks */
+#define CAP_NO_IV_PREFIX		BIT(8)
```

Comment:
you will need to put this api modification into its own commit. Then comes the mbedtls commit 

What about tc_shim commit as well? I wonder how you will be addressing ctr operator where (according to documentation) ivlen = keylen - ctrlen, so is of variable length.

Reply:
> put this api modification into its own commit
OK.

>What about tc_shim
These changes are required for another PR I'm working. I'll get to tc_shim later. Anyway each shim has to advertise its capabilities separately. So I guess it shouldn't be an issue.

>how you will be addressing ctr operator where (according to documentation) ivlen = keylen - ctrlen, so is of variable length
mbdtls_shim currently doesn't support CTR mode. One quick look at the tc_shim suggests that IV isn't prefixed in CTR mode, but need to take a closer look to be sure.

Reply:
Done.